src/scenechange/mod.rs

```diff
@@ -57,26 +56,26 @@ impl SceneChangeDetector {
     &mut self,
     previous_frame: Option<Arc<Frame<T>>>,
     frame_set: &[Arc<Frame<T>>],
-    mut input_frameno: u64,
+    input_frameno: u64,
     config: &EncoderConfig,
     inter_cfg: &InterConfig,
     keyframes: &mut BTreeSet<u64>,
   ) {
     let frame_set = match previous_frame {
-      Some(frame) => [frame].iter().chain(frame_set.iter()).cloned().collect(),
+      Some(frame) => [frame].iter().chain(frame_set.iter()).cloned().collect::<Vec<_>>(),
       None => {
         // The first frame is always a keyframe.
         keyframes.insert(0);
-        input_frameno += 1;
-        frame_set.to_owned()
+        return;
       }
     };
 
+    self.exclude_scene_flashes(&frame_set, input_frameno, inter_cfg);
```

Comment:
Correct me if I'm wrong, but I think right now if the max GOP size happens to land exactly in the middle of a flash, then the middle of a flash would get an I-frame, and then the end of the flash wouldn't? Do we want any special logic for this, e.g. inserting an I-frame right before the flash in this case (what happens if that's below min GOP size though)?

Reply:
I believe you're correct. I did not see any special code for that in x264 either (where this optimization is adapted from), but I think it would be valuable to add it. Worth doing in this PR, or making a separate issue to track it?

Reply:
I'd consider this later, but we really should document somewhere how this code works.